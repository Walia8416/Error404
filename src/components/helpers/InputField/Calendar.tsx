import {Calendar, CalendarList} from 'react-native-calendars';
import {Image, StyleSheet, Text, View} from 'react-native';
import React, {Dispatch, SetStateAction, useState} from 'react';

import CenterModel from '../models/CenterModel';
import {Colors} from '../../../constants/colors';
import Images from '../../../constants/icon';
import {Regular} from '../../../constants/Fonts';

interface CalendarProps {
  date?: string | Date;
  setSelectedDate: React.Dispatch<React.SetStateAction<object>>;
  open: boolean;
  close: Dispatch<SetStateAction<boolean>>;
  rangeSelection?: boolean;
}

interface RenderHeaderProp {
  date: {
    getDate: () => void;
    getDay: () => void;
    getUTCFullYear: () => void;
  };
  months?: number | string;
  datevalue: {
    day?: string;
  };
}

interface CustomArrowProp {
  direction: {};
}

interface DateProps {
  day?: string;
  month?: string;
  year?: string;
}
const monthArray: string[] = [
  'Jan',
  'Feb',
  'Mar',
  'Apr',
  'May',
  'Jun',
  'Jul',
  'Aug',
  'Sep',
  'Oct',
  'Nov',
  'Dec',
];

const RenderHeader = ({date, months, datevalue}: RenderHeaderProp) => {
  let day = date.getDate();
  let year = date.getUTCFullYear();
  let Day = datevalue?.day ? datevalue.day : day;
  let Mon = months ? monthArray[months - 1] : monthArray[new Date().getMonth()];
  return (
    <Text
      style={{
        fontFamily: Regular,
      }}>{`${Day} ${Mon} ${year}`}</Text>
  );
};

const CustomArrow = ({direction}: CustomArrowProp) => {
  return (
    <Image
      source={Images.backBlue}
      style={{
        width: 20,
        resizeMode: 'contain',
        height: 20,
        transform: [{rotate: direction === 'right' ? '180deg' : '0deg'}],
      }}
    />
  );
};

const CalendarComponent = ({
  close,
  date,
  open,
  rangeSelection,
  setSelectedDate,
}: CalendarProps) => {
  const [datevalue, setDateValue] = useState({});
  const [startValue, setStartValue] = useState({});
  const [endValue, setEndValue] = useState({});
  const [Month, setMonth] = useState('');
  const [count, setCount] = useState(1);
  const [orderError, setOrderError] = useState(false);

  const selectDay = (date: DateProps) => {
    setDateValue(date);
    setSelectedDate(date);
  };

  const isEmpty = (value: any) => {
    return (
      value && Object.keys(value).length === 0 && value.constructor === Object
    );
  };
  const rangeHandler = (date: {}) => {
    if (count % 2 == 0) {
      setEndValue(date);
      close(orderError ? true : false);
    } else {
      setStartValue(date);
    }
    setCount(count + 1);
  };
  React.useEffect(() => {
    if (!isEmpty(startValue) && !isEmpty(startValue)) {
      if (
        new Date(startValue?.dateString).getTime() <
        new Date(endValue?.dateString).getTime()
      ) {
        setOrderError(false);

        setSelectedDate({
          start: startValue,
          end: endValue,
        });
      } else {
        setOrderError(true);
      }
    }
  }, [startValue, endValue]);

  return (
    <CenterModel
      itemStyle={{height: 'auto'}}
      modelState={open}
      setmodelState={close}
      trasparent={true}>
      <Calendar
        testID="calender"
        calendarWidth={320}
        current={date}
        disableAllTouchEventsForDisabledDays={true}
        disableArrowLeft={false}
        disableArrowRight={false}
        disableMonthChange={false}
        enableSwipeMonths={true}
        firstDay={1}
        futureScrollRange={50}
        hideArrows={false}
        hideDayNames={false}
        hideExtraDays={false}
        horizontal={true}
        maxDate={'2030-05-30'}
        minDate={new Date()}
        monthFormat={'yyyy MM'}
        markedDates={
          rangeSelection
            ? {
                [`${startValue?.dateString}`]: {
                  selected: true,
                  selectedColor: Colors.Crimson,
                },
                [`${endValue?.dateString}`]: {
                  selected: orderError ? false : true,
                  selectedColor: Colors.link,
                },
              }
            : {
                [`${datevalue?.dateString}`]: {
                  selected: true,
                  selectedColor: Colors.Crimson,
                },
              }
        }
        onDayPress={day => {
          rangeSelection ? rangeHandler(day) : selectDay(day);
        }}
        onVisibleMonthsChange={months => {
          setMonth(months[0].month);
        }}
        onPressArrowLeft={subtractMonth => subtractMonth()}
        onPressArrowRight={addMonth => addMonth()}
        pagingEnabled={true}
        pastScrollRange={50}
        renderArrow={direction => <CustomArrow direction={direction} />}
        renderHeader={date => {
          return (
            <RenderHeader date={date} months={Month} datevalue={datevalue} />
          );
        }}
        scrollEnabled={true}
        showScrollIndicator={true}
      />
    </CenterModel>
  );
};

export default CalendarComponent;

const styles = StyleSheet.create({});
