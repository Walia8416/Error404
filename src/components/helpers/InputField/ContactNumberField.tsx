import React from 'react';
import {Picker} from '@react-native-picker/picker';
import {details} from './data';
import {Colors} from '../../../constants/colors';

interface Props {
  selectedCode: string; //counry code
  setSelectedCode: (args: string) => {};
  code: [{label: string; value: string}]; //array of counry code
  enable?: boolean;
}

const ContactNumberField = (props: Props) => {
  return (
    <Picker
      testID="react.picker.test"
      selectedValue={props.selectedCode}
      onValueChange={(itemValue: string, itemIndex: number) => {
        props.setSelectedCode(itemValue);
      }}
      mode="dropdown"
      enabled={!props.enable}
      style={{
        width: '45%',
        color: !props.enable ? Colors.DimGray : Colors.inputBorder,
      }}>
      {details.map(item => (
        <Picker.Item
          label={item.dial_code}
          value={item.dial_code}
          key={item.dial_code}
          style={{width: 20, color:Colors.FeatherWhite,fontSize:15}}
        />
      ))}
    </Picker>
  );
};

export default ContactNumberField;
