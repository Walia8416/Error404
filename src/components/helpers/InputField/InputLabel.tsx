import {StyleSheet, Text, View} from 'react-native';

import {Colors} from '../../../constants/colors';
import {RFValue} from 'react-native-responsive-fontsize';
import React from 'react';

interface Props {
  title: string;
}

const InputLabel: React.FC<Props> = ({title}: Props) => {
  return (
    <View style={styles.title}>
      <Text style={styles.text}>{title}</Text>
    </View>
  );
};

export default InputLabel;

const styles = StyleSheet.create({
  title: {
    textAlign: 'center',
    paddingVertical: 5,
    paddingLeft: 2,
  },
  text: {
    color: Colors.text,
    fontSize: RFValue(12),
  },
});
