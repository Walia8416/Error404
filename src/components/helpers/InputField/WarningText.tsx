import {Colors} from '../../../constants/colors';
import {Image} from 'react-native';
import Images from '../../../constants/icon';
import React from 'react';
import styled from 'styled-components/native';

const WarnText = styled.Text`
  color: ${Colors.danger};
  padding-left: 5px;
`;
const WarnView = styled.View`
  flex-direction: row;
  align-items: center;
`;

export const WarningText: React.FC<{title: string}> = ({title}) => {
  return (
    <WarnView testID="warningtext">
      {title !== '' && (
        <Image
          resizeMode="contain"
          source={Images.info}
          style={{width: 15, height: 15}}
        />
      )}
      <WarnText>{title}</WarnText>
    </WarnView>
  );
};
