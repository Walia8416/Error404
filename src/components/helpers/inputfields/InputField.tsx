import {
  ColorValue,
  Image,
  KeyboardTypeOptions,
  NativeSyntheticEvent,
  Text,
  TextInput,
  TextInputEndEditingEventData,
  View,
} from 'react-native';
import {Colors} from '../../../constants/colors';
import {
  Control,
  Controller,
  FieldValues,
  RegisterOptions,
  useForm,
} from 'react-hook-form';

import Images from '../../../constants/icon';
import React from 'react';

interface props {
  placeholder?: string;
  width?: number | string;
  height?: number | string;
  borderradius?: number | string;
  bordercolor?: string;
  onError?: boolean;
  onChangeText?: ((text: string) => void) | undefined;
  keyboardType?: KeyboardTypeOptions | undefined;
  passwordrules?: string | null | undefined;
  secure?: boolean | undefined;
  maxlength?: number | undefined;
  numberoflines?: number | undefined;
  onendediting?:
    | ((e: NativeSyntheticEvent<TextInputEndEditingEventData>) => void)
    | undefined;
  placeholdertextcolor?: ColorValue | undefined;
  ref?: React.LegacyRef<TextInput> | undefined;
  testID?: string | undefined;
  textalign?: 'center' | 'left' | 'right' | undefined;
  textalignvertical?: 'center' | 'auto' | 'top' | 'bottom' | undefined;
  text?: string;
  name?: string | `${string}.${string}` | `${string}.${number}`;
  controls?: Control<FieldValues> | undefined;
  rules?: Omit<RegisterOptions, 'valueAsNumber' | 'valueAsDate' | 'setValueAs'>;
  errorText?: boolean | undefined;
}

const InputFeild: React.FC<props> = ({
  width,
  height,
  placeholder,
  borderradius,
  bordercolor,
  onError,
  onChangeText,
  keyboardType,
  passwordrules,
  secure,
  maxlength,
  numberoflines,
  onendediting,
  placeholdertextcolor,
  ref,
  testID,
  textalign,
  textalignvertical,
  text,
  name,
  controls,
  rules,
  errorText,
}) => {
  const [onActive, setActive] = React.useState(false);
  const bordercolorSet = onError
    ? '#e23b3b'
    : onActive
    ? Colors.x
    : bordercolor
    ? bordercolor
    : '#c4d7e7';

  const {control} = useForm();

  return (
    <View
      style={{
        width: width ? width : 312,
        height: height ? height : 72,
        justifyContent: 'center',
        alignItems: 'center',
        // marginBottom: 32,
        alignSelf: 'center',
      }}>
      <Text
        style={{
          fontFamily: 'Inter-Regular',
          alignSelf: 'flex-start',
          marginBottom: 8,
          color: '#839fc0',
          fontSize: 12,
        }}>
        {text}
      </Text>
      <Controller
        rules={rules}
        control={controls ? controls : control}
        name={name ? name : 'default'}
        render={({field: {onChange, onBlur, value}, fieldState, formState}) => (
          <TextInput
            style={{
              borderColor: bordercolorSet,
              width: width ? width : 312,
              height: height ? height : 46,
              color: '#153059',
              borderWidth: 1,
              borderRadius: borderradius ? borderradius : 0,
              paddingLeft: 16,
              fontSize: 14,
              fontFamily: 'Inter-Regular',
            }}
            placeholder={placeholder ? placeholder : 'Enter Some thing'}
            onFocus={() => setActive(true)}
            onBlur={onBlur}
            onChangeText={onChange}
            keyboardType={keyboardType}
            passwordRules={passwordrules}
            secureTextEntry={secure}
            autoCorrect={false}
            maxLength={maxlength}
            numberOfLines={numberoflines}
            onEndEditing={onendediting}
            placeholderTextColor={placeholdertextcolor}
            ref={ref}
            testID={testID}
            textAlign={textalign}
            textAlignVertical={textalignvertical}
            value={value}
            // inlineImageLeft={null}
            // inlineImagePadding
          />
        )}
      />
      {errorText ? (
        <View
          style={{
            height: 18,
            width: 210,
            marginTop: 12,
            flexDirection: 'row',
            alignSelf: 'flex-start',
          }}>
          <Image
            source={Images.info}
            style={{width: 16, height: 16}}
            resizeMode={'contain'}
          />
          <Text
            style={{
              fontFamily: 'Inter-Regular',
              fontSize: 12,
              textAlignVertical: 'center',
              marginLeft: 8,
              width: 250,
              color: '#e23b3b',
            }}>
            {errorText}
          </Text>
        </View>
      ) : (
        <View
          style={{
            height: 18,
            width: 210,
            marginTop: 12,
            flexDirection: 'row',
            alignSelf: 'flex-start',
          }}
        />
      )}
    </View>
  );
};

export default InputFeild;
