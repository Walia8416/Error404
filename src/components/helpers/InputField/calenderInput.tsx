import {
  Control,
  DeepMap,
  FieldError,
  FieldValues,
  UseFormSetValue,
} from 'react-hook-form';
import React, {useState} from 'react';

import CalendarComponent from './Calendar';
import InputField from './InputField';
import {View} from 'react-native';

interface CalenderProps {
  control: Control<FieldValues>;
  errors: DeepMap<FieldValues, FieldError>;
  name: string;
  placeholder?: string;
  setValue: UseFormSetValue<FieldValues>;
  title?: string;
  rule?: {};
  half?: boolean;
  styles?: {};
  calenderIcon?: string | NodeRequire;
}

interface DateProps {
  day?: string;
  month?: string;
  year?: string;
}

const CalenderInput = ({
  calenderIcon,
  control,
  errors,
  half,
  name,
  placeholder,
  rule,
  setValue,
  styles,
  title,
}: CalenderProps) => {
  const [datevalue, setDateValue] = useState(new Date());
  const [modalopen, setmodalopen] = useState<boolean>(false);

  const selectDay = (date: DateProps) => {
    const {day, month, year} = date;

    setValue(name, `${day}/${month}/${year}`);
    setDateValue(date);
    setmodalopen(false);
  };

  return (
    <View style={styles}>
      <InputField
        calender={true}
        calenderFun={() => setmodalopen(!modalopen)}
        calenderIcon={calenderIcon}
        control={control}
        errors={errors[name]}
        half={half}
        name={name}
        placeholder={placeholder}
        rules={rule}
        testID="calenderInput.test"
        title={title}
      />
      <CalendarComponent
        date={datevalue}
        setSelectedDate={selectDay}
        open={modalopen}
        close={setmodalopen}
      />
    </View>
  );
};

export default CalenderInput;
