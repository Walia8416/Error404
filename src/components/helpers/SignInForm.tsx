import {Button, View} from 'react-native';

import InputField from './InputField/InputField';
import React from 'react';
import {useForm} from 'react-hook-form';
const SignInForm: React.FC = () => {
  const {
    control,
    handleSubmit,
    formState: {isValid, errors},
  } = useForm({
    mode: 'onChange', //used to check error in realtime
    shouldFocusError: true, // focus input field after submit if it is not following required rule of input field
  }); // react hook form main state
  // form Submit function
  const onSubmit = (data: object) => {};
  // Required props for InputField Component are (title, name, placeholder, control, errors, rules, testID)
  return (
    <View
      style={{
        paddingTop: 0,
        justifyContent: 'center',
        alignItems: 'center',
        marginBottom: 12,
      }}>
      {/* email input */}
      <InputField
        testID="email.test"
        title="User name"
        name="Username"
        control={control}
        errors={errors?.Username}
        placeholder="E-mail address"
        rules={{
          required: true,
          type: 'email',
          pattern: {
            value: /\S+@\S+.\S+/,
            message: 'Invalid email format.Please try again.',
          },
        }}
      />
      {/* Password Input */}
      <InputField
        testID="password.test"
        title="Password"
        name="password"
        control={control}
        errors={errors?.password}
        placeholder="Enter your password"
        pass={true}
        rules={{
          required: true,
          type: 'password',
          minLength: {
            value: 2,
            message: '',
          },
        }}
      />
      {/* <Button title="Submit" onPress={handleSubmit(onSubmit)} /> */}
    </View>
  );
};
export default SignInForm;
