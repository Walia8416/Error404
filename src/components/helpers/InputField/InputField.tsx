import { Controller, useForm } from 'react-hook-form';
import { Image, TextInput, TouchableOpacity, View } from 'react-native';
import { Input, InputBox } from '../../../styled_components/textfields';

import CircleTouch from '../CircleTouch';
import { Colors } from '../../../constants/colors';
import ContactNumberField from './ContactNumberField';
import Images from '../../../constants/icon';
import InputLabel from './InputLabel';
import React from 'react';
import { WarningText } from './WarningText';

// Styled Component

// Custom Component

//images

interface Props {
  code?: any; // country code
  containerStyle?: {}; //styling of container
  control?: any; // control prop for inputfield from react hook form
  defaultValue?: string;
  errors?: { message: string }; // error message
  half?: boolean; //half width
  Inputvalue?: string; // can give value direct
  keyboardType?: string; //keyboardtype
  maxLength?: number;
  multiline?: boolean;
  name: string; // name of the input field (required)
  numberOfLines?: number;
  pass?: boolean;
  phone?: boolean;
  placeholder?: string; // placeholder for input
  rules?: object; // condition you want to apply for input
  selectedCode?: any; //selected country code
  setSelectedCode?: any; // selecting code function
  testID?: string; // testid for tesing component
  title?: string; // label of the input field
  calender?: boolean;
  calenderFun?: () => void;
  calenderIcon?: string | NodeRequire;
  boxStyle?: {};
  inputStyles?: {};
  inputRef?: React.LegacyRef<TextInput> | undefined;
  titleimage?: string | NodeRequire;
  titleImageStyle?: {};
  titleImagePress?: () => void;
  disable?: boolean;
  rightside?: any;
  china?: boolean;
}
const InputField: React.FC<Props> = (props: React.PropsWithChildren<Props>) => {
  // password icon state
  const [seenPass, setSeenpass] = React.useState(true);
  const [chinaRules, setChinaRules] = React.useState({
    required: true,
    type: 'phone',
    minLength: {
      value: 11,
      message: 'Phone must have 11 digits.Please try again.',
    },
    maxLength: {
      value: 11,
      message: 'Phone must not have more than 11 digits.Please try again.',
    },
  })
  const { control } = useForm();
  // inputfield border warning style variable
  let warningHandler = props.errors ? Colors.danger : Colors.success;
  return (
    <View
      style={[
        { marginBottom: 10, width: props.half === true ? '45%' : '100%' },
        props.containerStyle,
      ]}>
      {/* inputfield title component (InputLabel) */}
      <View
        style={{
          flexDirection: 'row',
          alignItems: 'center',
        }}>
        {props.title && <InputLabel title={props.title} />}
        {props.titleimage && (
          <TouchableOpacity onPress={props.titleImagePress}>
            <Image source={props.titleimage} style={props.titleImageStyle} />
          </TouchableOpacity>
        )}
      </View>
      <Controller
        control={props.control ? props.control : control}
        render={({ field: { onChange, onBlur, value } }) => (
          // inputfield cover component (InputBox)
          <InputBox
            style={[
              {
                borderColor:
                  value === '' && !props.errors
                    ? Colors.inputBorder
                    : warningHandler,
              },
              props.boxStyle,
            ]}>
            {props.phone === true && (
              //country code picker component
              <ContactNumberField
                code={props.code} // array of country code from backend
                selectedCode={props.selectedCode} // selected code  from picker to parent component
                setSelectedCode={props.setSelectedCode} // selecting country code from picker to parent component
                enable={props.disable}
              />
            )}

            {/* TextInput Styled component (Input) */}
            <Input
              testID={props.testID}
              onBlur={onBlur}
              onChangeText={onChange}
              value={props.Inputvalue ? props.Inputvalue : value}
              placeholderTextColor={Colors.placeHolder}
              editable={props.disabled}
              // used to prevent font change while setting inputfield for password : issue with secureTextEntry
              // ref={ref =>
              //   ref &&
              //   ref.setNativeProps({style: {fontFamily: 'Roboto-Regular'}})
              // }
              ref={props.inputRef}
              secureTextEntry={props.pass && seenPass ? true : false}
              style={props.inputStyles}
              editable={!props.disable}
              style={{
                color:
                  !props.disable === true ? Colors.DimGray : Colors.inputBorder,
              }}
              // for addition of standard props fields for input
              {...props}
            />
            {/* Visible and not visible icon button for password field */}
            {props.pass && (
              <TouchableOpacity
                testID="eye.test"
                activeOpacity={0.6}
                onPress={() => setSeenpass(!seenPass)}>
                {seenPass && (
                  <Image
                    testID="closeEye.test"
                    resizeMode="center"
                    source={Images.eyeOff}
                    width={15}
                    height={15}
                  />
                )}
                {!seenPass && (
                  <Image
                    testID="openEye.test"
                    resizeMode="center"
                    source={Images.eyeOn}
                    width={15}
                    height={15}
                  />
                )}
              </TouchableOpacity>
            )}
            {props.calender && (
              <CircleTouch onPress={props.calenderFun}>
                <Image
                  source={
                    props.calenderIcon ? props.calenderIcon : Images.Calender
                  }
                />
              </CircleTouch>
            )}
            {props.rightside}
          </InputBox>
        )}
        name={props.name}



        defaultValue={props.defaultValue ? props.defaultValue : ''}
        rules={props.china && props.phone ? chinaRules : props.rules}
      />
      {/* InputField warning Text styled component  */}
      {props.errors && !!props.errors?.message && (
        <WarningText title={props.errors.message} />
      )}
    </View>
  );
};
export default InputField;
