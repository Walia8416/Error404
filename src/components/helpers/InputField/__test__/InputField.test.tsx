import 'react-native';
import 'react-hook-form';

import {FormProvider, useForm} from 'react-hook-form';
import {fireEvent, render} from '@testing-library/react-native';

import {Calendar} from 'react-native-calendars';
import ContactNumberField from '../ContactNumberField';
import InputField from '../InputField';
import InputLabel from '../InputLabel';
import React from 'react';
import SignInForm from '../../SignInForm';
import {WarningText} from '../WarningText';
import {create} from 'react-test-renderer';

it('renders form', () => {
  const input = create(
    <InputField title="User name" name="email" placeholder="E-mail address" />,
  );
  expect(input).toMatchSnapshot();
});

it('renders form label', () => {
  render(<InputLabel title="User nam" />);
});

it('Check input picker rendered', () => {
  let code: [{label: string; value: string}];
  code = [
    {
      label: '+91',
      value: '+91',
    },
  ];
  render(<SignInForm />);

  render(
    <ContactNumberField
      code={code}
      selectedCode="+91"
      setSelectedCode={jest.fn()}
    />,
  );
});

it('check warning text', () => {
  const {getByText} = render(<WarningText title="warningtext" />);
  expect(getByText('warningtext')).toBeTruthy();
});

it('picker for country code', () => {
  render(
    <ContactNumberField
      code={[{label: '+1', value: '+1'}]}
      selectedCode="+1"
      setSelectedCode={jest.fn()}
    />,
  );
});

const CustomComponent = () => {
  const methods = useForm();
  return (
    <FormProvider {...methods}>
      <InputField
        errors={{message: 'Invalid mail'}}
        testID={'input'}
        placeholder={'hello bettervitty'}
        name="Name"
        Inputvalue="lorem"
        control={methods.control}
        rules={{
          required: true,
          type: 'email',
          pattern: {
            value: /\S+@\S+.\S+/u,
            message: 'Invalid email format.Please try again.',
          },
        }}
        pass={true}
      />
    </FormProvider>
  );
};
it('textField render', async () => {
  const {getByTestId} = render(<CustomComponent />);
  expect(getByTestId('closeEye.test')).toBeTruthy();
  await fireEvent.press(getByTestId('eye.test'));
  expect(getByTestId('openEye.test')).toBeTruthy();
  expect(getByTestId('warningtext')).toBeTruthy();
});

it('render contact field', () => {
  const inputfield = create(
    <ContactNumberField
      code={[{label: '+1', value: '+1'}]}
      selectedCode="+1"
      setSelectedCode={jest.fn()}
    />,
  );
  const input = inputfield.root.findByProps({
    testID: 'react.picker.test',
  }).props;
  input.onValueChange();
});

//calender
const CalenderComponent = () => {
  const methods = useForm();
  return (
    <FormProvider {...methods}>
      <InputField
        errors={{message: 'Invalid mail'}}
        testID={'calenderInput.test'}
        placeholder={'hello bettervitty'}
        name="Name"
        control={methods.control}
        rules={{
          required: true,
        }}
        calender={true}
        calenderFun={jest.fn()}
      />
    </FormProvider>
  );
};
it('calneder render', () => {
  const Calender = create(<CalenderComponent />);
  const Props = Calender.root.findAllByProps({testID: 'calenderInput.test'})[0]
    .props;
  // expect(getByTestId('stateTest')).toBeTruthy();
  Props.calenderFun(jest.fn());
});
