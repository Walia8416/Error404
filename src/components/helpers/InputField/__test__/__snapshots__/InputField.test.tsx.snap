// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 10,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "paddingLeft": 2,
          "paddingVertical": 5,
          "textAlign": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#415f8b",
            "fontSize": 24,
          }
        }
      >
        User name
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "#c4d7e7",
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "fontFamily": "Inter-Regular",
          "height": 96,
          "justifyContent": "space-between",
          "marginTop": 5,
          "paddingBottom": 0,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 0,
        },
        Object {
          "borderColor": "#c4d7e7",
        },
        undefined,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      name="email"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="E-mail address"
      placeholderTextColor="#839fc0"
      rejectResponderTermination={true}
      secureTextEntry={false}
      style={
        Array [
          Object {
            "color": "#415F8B",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
          },
        ]
      }
      title="User name"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
